src/packet/ipv6.rs

```diff
@@ -32,6 +32,84 @@ pub struct Ipv6 {
     payload: Vec<u8>,
 }
 
+/// Represents an IPv6 Packet pseudo header
+#[packet]
+pub struct Ipv6PseudoHeader {
+    #[construct_with(u16, u16, u16, u16, u16, u16, u16, u16)]
+    source: Ipv6Addr,
+    /// RFC 2460:
+    /// If the IPv6 packet contains a Routing header, the Destination Address used in the
+    /// pseudo-header is that of the final destination. At the originating node, that address will
+    /// be in the last element of the Routing header; at the recipient(s), that address will be in
+    /// the Destination Address field of the IPv6 header.
+    #[construct_with(u16, u16, u16, u16, u16, u16, u16, u16)]
```

Comment:
If this is a direct quote, please denote it as such using markdown quote syntax:
```
> text
> more text
```

Also, please add a URL for the RFCs, and change the formatting to (note the extra line and the lack of a `:`):

```rust
/// RFC 1234
/// <link to RFC>
///
/// > the quote
```

I'd also like a newline between fields and comments for the next field, if the comments are multiple lines (which they are here).